### PR TITLE
bus/nes: Fixed a Boogerman bootleg.

### PR DESCRIPTION
**Pull request by 0kmg**
- Reorganized sglionk (Super Game bootleg games) around lookup tables for its register/address scrambling.
- Implemented Boogerman it terms of sglionk, since it is functionally identical with different scrambling permutations.

New working software list additions (nes.xml)
-----------------------------------
The Lion King (pirate, alt 2) [Consolethinks]

Software list items promoted to working (nes.xml)
---------------------------------------
Boogerman (Rus)

### CODE REVIEW

**Comment by 0kmg** [comment]
I should note that I've commented out the old code for Boogerman instead of deleting it. Not sure if it will be needed. There are 7 other games that use the same cart slot. None of them ever showed much more than a grey screen and are different enough from Boogerman to not really be compatible. Sorting them out will hopefully be a future PR.

File: src/devices/bus/nes/mmc3_clones.cpp [hunk 1]
```diff
@@ -109,13 +109,33 @@ nes_8237_device::nes_8237_device(const machine_config &mconfig, const char *tag,
...
-nes_sgboog_device::nes_sgboog_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
-	: nes_txrom_device(mconfig, NES_SG_BOOG, tag, owner, clock), m_mode(0)
+nes_sglionk_device::nes_sglionk_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
+	: nes_sglionk_device(mconfig,
+			     NES_SG_LIONK,
+			     tag,
+			     owner,
+			     clock,
+			     (const u8 []) {0, 3, 1, 5, 6, 7, 2, 4},
+			     (const u16 []) {0xa001, 0xa000, 0x8000, 0xc000, 0x8001, 0xc001, 0xe000, 0xe001})
```
Comment (cuavas) [inline src/devices/bus/nes/mmc3_clones.cpp:127]: Is this actually safe?  Strictly speaking, aren’t these temporaries?  I get the feeling this works by coincidence due to how the compiler has implemented it.

Comment (0kmg) [reply src/devices/bus/nes/mmc3_clones.cpp:127]: Damn, my C++-fu is not strong enough to answer this question. In C++11 a temporary object's lifetime is extended if bound to a const reference, but I'm not sure if the same rule applies portably to pointers in this case.

Comment (cuavas) [reply src/devices/bus/nes/mmc3_clones.cpp:127]: Yeah, but the const reference it’s bound to there would just be the argument to the base class constructor, which only has the lifetime of the base class constructor invocation, so it wouldn’t help you.

Comment (0kmg) [reply src/devices/bus/nes/mmc3_clones.cpp:127]: Well regardless of whether it's kosher, I can't really defend the choices here. I guess I was trying hard to be creative? I'm submitting a more traditional, staid patch.

**Pull request merged**